File: checkio_js/electronic_station/javascript/isAscending.js
import assert from "assert";

function isAscending(values) {
    for (let i = 0; i<=values.length; i++){
        if (values[i] > values[i+1]){
            return false
        }
    }

    return values.length <= 1 ? true :
        new Set(values).size === 1 ? false :
        true
}

console.log('Example:');
console.log(isAscending([-5, 10, 99, 123456]));

// These "asserts" are used for self-checking
assert.equal(isAscending([-5, 10, 99, 123456]), true);
assert.equal(isAscending([99]), true);
assert.equal(isAscending([4, 5, 6, 7, 3, 7, 9]), false);
assert.equal(isAscending([]), true);
assert.equal(isAscending([1, 1, 1, 1]), false);
assert.equal(isAscending([-1, 0, 1]), true);

console.log("Coding complete? Click 'Check' to earn cool rewards!");

File: checkio_js/electronic_station/python/isAscending.py


def isAscending(values):
    return False if len(values) > 1 and len(set(values)) == 1 else values == sorted(values)

if __name__ == '__main__':
    print('Example:')
    print(isAscending([-5, 10, 99, 123456]))

# These "asserts" are used for self-checking
assert isAscending([-5, 10, 99, 123456]) == True
assert isAscending([99]) == True
assert isAscending([4, 5, 6, 7, 3, 7, 9]) == False
assert isAscending([]) == True
assert isAscending([1, 1, 1, 1]) == False

print("Coding complete? Click 'Check' to earn cool rewards!")
